x/twap/logic_test.go

```diff
@@ -42,76 +43,287 @@ func newExpRecord(accum0, accum1 sdk.Dec) types.TwapRecord {
 	}
 }
 
-func TestRecordWithUpdatedAccumulators(t *testing.T) {
+func (s *TestSuite) TestNewTwapRecord() {
+	// prepare pool before test
+	poolId := s.PrepareBalancerPoolWithCoins(defaultTwoAssetCoins...)
+
+	tests := map[string]struct {
+		poolId        uint64
+		denom0        string
+		denom1        string
+		expectedErr   bool
+		expectedPanic bool
+	}{
+		"denom with lexicographical order": {
+			poolId,
+			denom1,
+			denom0,
+			false,
+			false,
+		},
+		"denom with non-lexicographical order": {
+			poolId,
+			denom0,
+			denom1,
+			false,
+			false,
+		},
```

Comment:
These are backwards. The first one is in non lexicographical order and the second one is in lexicographical order.